pkg/discovery/discovery.go

```diff
@@ -360,12 +361,20 @@ func (d *Discoverer) discoverVolumesAtPath(class string, config common.MountConf
 				discoErrors = append(discoErrors, fmt.Errorf("path %q of filesystem mode cannot be used to create block volume", filePath))
 				continue
 			}
-			// Validate that this path is an actual mountpoint
-			if _, isMntPnt := mountPointMap[filePath]; isMntPnt == false {
-				discoErrors = append(discoErrors, fmt.Errorf("path %q is not an actual mountpoint", filePath))
-				continue
+			// TODO(mauriciopoppe): skipping the mountpoint verification in Windows, unlike Linux there's no /proc/mounts file to check
```

Comment:
I'm gonna address this TODO in the e2e PR

Reply:
I'm not sure if this is possible, the discovery directory contains only symlinks which when dereferenced give a volumeId like `\\?\Volume{452e318a-5cde-421e-9831-b9853c521012}\` which can be understood only in the context of the host.

I'm afraid I didn't answer your question though, I'm not very familiar with how the PV works under the hood i.e. what args will be sent to the container runtime in Windows, do you have a suggestion to test this scenario?

Reply:
It sounds like in windows 2) is done first and then 1) uses a symlink into the discovery directory. In that case, should we verify here that the path is a symlink?

Reply:
yes that's a good point, right now if I remember correctly an error is collected if the path to open is not a symlink, agreed we should check if it's a symlink to consider it as the source of a PV

Reply:
the error is collected indirectly in the call GetFsCapacityByte, the first step is to dereference the symlink and if it's not a symlink then it'll fail

```
			if runtime.GOOS == "windows" {
				// the path to use is the one sent to CSI Proxy and must be in the context of the host
				// not the container, filePath is in the context of the container
				volumePathInHost := filepath.Join(config.HostDir, file)
				capacityByte, err = d.VolUtil.GetFsCapacityByte(volumePathInHost)
```

Reply:
Is the error message clear on the problem?

Reply:
I created a symlink pointing to a volume `ssd0` and a directory `fake-ssd1` in the discovery dir, in the logs I saw:

```
E1206 23:51:23.874612    1568 discovery.go:222] Failed to discover local volumes: 1 error(s) while discovering volumes: [path "c:\\mnt\\disks\\fake-ssd1" fs stats error: rpc error: code = Unknown desc = error getting the volume │
│  for the mount c:\mnt\disks\fake-ssd1, internal error error getting volume from mount. cmd: (Get-Item -Path c:\mnt\disks\fake-ssd1).Target, output: , error: <nil>] 
```

I think we have the call [`isSymlink`](https://github.com/kubernetes-csi/csi-proxy/blob/master/docs/apis/filesystem_v1.md) in CSI Proxy that we can use for a better error , adding it...

Reply:
Added a couple of new commits, the last one is using CSI Proxy's IsSymlink https://github.com/kubernetes-csi/csi-proxy/blob/4c23ef2be1fbb3607619c71f3c4289b42e05342a/pkg/os/filesystem/api.go#L93 to check if the files in the discovery dir are likely mount points

The new error looks like this:

```
1207 05:24:49.494358    3028 cache.go:55] Added pv "local-pv-20666959" to cache                                                                                                                                                    I1207 05:24:49.494939    3028 discovery.go:459] Created PV "local-pv-20666959" for volume at "c:\\mnt\\disks\\ssd0" 
E1207 05:24:49.494939    3028 discovery.go:222] Failed to discover local volumes: 1 error(s) while discovering volumes: [path "c:\\mnt\\disks\\fake-ssd1" is likely not a mount point: <nil>] 
I1207 05:24:49.508014    3028 cache.go:64] Updated pv "local-pv-20666959" to cache
```